lib/src/json_schema/constants.dart

```diff
@@ -37,6 +37,14 @@
 //     THE SOFTWARE.
 
 class JsonSchemaValidationRegexes {
+  // Spec: https://datatracker.ietf.org/doc/html/rfc3339#appendix-A
+  static RegExp duration = RegExp(
+      r'^P(\d{1,}W|T(\d{1,}H(\d{1,}M(\d{1,}S)?)?|\d{1,}M(\d{1,}S)?|\d{1,}S)|(\d{1,}D|\d{1,}M(\d{1,}D)?|\d{1,}Y(\d{1,}M(\d{1,}D)?)?)(T(\d{1,}H(\d{1,}M(\d{1,}S)?)?|\d{1,}M(\d{1,}S)?|\d{1,}S))?)$');
```

Comment:
I built this in the most naive possible way from the grammar. If you're able to do this more cleverly, please let me know.